base/iterators.jl

```diff
@@ -1396,19 +1397,24 @@ Stacktrace:
 [...]
 
 julia> only(('a', 'b'))
-ERROR: ArgumentError: Tuple contains 2 elements, must contain exactly 1 element
+ERROR: ArgumentError: Tuple contains multiple elements, must contain exactly 1 element
+Stacktrace:
+[...]
+
+julia> only(('a', 'b'), "my error message")
```

Comment:
```suggestion
only(('a', 'b'); hint = "my error hint")
```

Reply:
@bramtayl I think that macros are generally more flexible, but I think that would be an addition outside the scope here.

@mcabbott Good point about performance. I think that @DilumAluthge 's changes are constructive in the spirit of avoiding unnecessary interpolation.

@DilumAluthge Regarding your changes to move to keyword arguments, is there actually a significant win over a default value approach? I can't see that this is a hotbed of multiple dispatch, although one could imagine that it might become such (for instance, passing in a fully constructed exception to raise instead of the normal ArgumentException) but I can't see a big difference either way. If you feel there even a modest wind blowing toward the keyword approach, I am fine with it. I would tune the example slightly. I might also go further by adding message and hint.

Reply:
Why does making it a keyword change the interpolation cost? The user's string is still made before the function is called.

What could perhaps avoid that without a macro is to accept e.g. `only(xs, "expected ", typeof(x), " to have one element ")`, crudely:
``` 
only(things, message...) = length(things)==1 ? things[1] : error(string(message...))
```
But that also seems a bit weird, and  you have to remember to write it that way.